thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/events/HolidayEventsLoader.java

```diff
@@ -0,0 +1,337 @@
+    List<EventDTO> existingEvents = eventDAO.findEventsBetweenTimeRange(EventType.HOLIDAY.toString(), start, end);
+    Map<HolidayEvent, EventDTO> existingHolidayEventToEventDTO = new HashMap<>();
+    Map<HolidayEvent, Set<String>> existingHolidayEventToCountryCodes = new HashMap<>();
+
+    // Map existing event DTO to the tuple of HolidayEvent and country code set
+    for (EventDTO existingEventDTO : existingEvents) {
+      HolidayEvent holidayEvent = new HolidayEvent(existingEventDTO);
+      if (!existingHolidayEventToCountryCodes.containsKey(holidayEvent)) {
+        existingHolidayEventToCountryCodes.put(holidayEvent, new HashSet<String>());
+      }
+      if (existingEventDTO.getTargetDimensionMap() != null
+          && existingEventDTO.getTargetDimensionMap().get("countryCode") != null) {
+        existingHolidayEventToCountryCodes.get(holidayEvent)
+            .addAll(existingEventDTO.getTargetDimensionMap().get("countryCode"));
+      }
+      existingHolidayEventToEventDTO.put(holidayEvent, existingEventDTO);
+    }
+
+    for (Map.Entry<HolidayEvent, Set<String>> entry : newHolidayEventToCountryCodes.entrySet()) {
+      HolidayEvent newHolidayEvent = entry.getKey();
+      Set<String> newCountryCodes = entry.getValue();
+
+      if (existingHolidayEventToCountryCodes.containsKey(newHolidayEvent)) {
+        if (existingHolidayEventToCountryCodes.get(newHolidayEvent).addAll(newCountryCodes)) {
+          // If the holiday is already exist and the country codes are different, merge with new country codes and update the database
+          Map<String, List<String>> targetDimensionMap = new HashMap<>();
+          targetDimensionMap.put("countryCode", new ArrayList<>(newCountryCodes));
+
+          EventDTO eventDTO = existingHolidayEventToEventDTO.get(newHolidayEvent);
+          eventDTO.setTargetDimensionMap(targetDimensionMap);
+          eventDAO.update(eventDTO);
+        }
+      } else {
+        // If the holiday is not exist, insert it into the database
+        EventDTO eventDTO = new EventDTO();
+        eventDTO.setName(entry.getKey().getName());
+        eventDTO.setEventType(entry.getKey().getEventType());
+        eventDTO.setStartTime(entry.getKey().getStartTime());
+        eventDTO.setEndTime(entry.getKey().getEndTime());
+
+        Map<String, List<String>> targetDimensionMap = new HashMap<>();
+        targetDimensionMap.put("countryCode", new ArrayList<>(entry.getValue()));
+        eventDTO.setTargetDimensionMap(targetDimensionMap);
+        eventDAO.save(eventDTO);
+      }
+    }
+  }
+
+  private String getCountryCode(Event holiday) {
+    String countryName = holiday.getCreator().getDisplayName().substring(12);
+    for (Locale locale : Locale.getAvailableLocales()) {
+      if (locale.getDisplayCountry().equals(countryName)) {
+        return locale.getCountry();
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Fetch all holidays from Google Calendar API.
```

Comment:
Add the condition of holidays that will be returned by this method: a description like holidays whose end time GE start and start time LT end (assuming you are using end time exclusive).